src/Rule/FilenameUsesUnderscoresOnly.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of DOCtor-RST.
+ *
+ * (c) Oskar Stark <oskarstark@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Rule;
+
+use App\Annotations\Rule\Description;
+use App\Annotations\Rule\InvalidExample;
+use App\Annotations\Rule\ValidExample;
+use App\Value\Lines;
+use App\Value\RuleGroup;
+use function Symfony\Component\String\u;
+
+/**
+ * @Description("Ensures a filename uses only underscores (`_`).")
+ * @InvalidExample("custom-extensions.rst")
+ * @ValidExample("custom_extensions.rst")
+ * @ValidExample("_custom_extensions.rst")
+ */
+final class FilenameUsesUnderscoresOnly extends AbstractRule implements Rule
+{
+    public static function getGroups(): array
+    {
+        return [
+            RuleGroup::Sonata(),
+            RuleGroup::Symfony(),
+        ];
+    }
+
+    public function check(Lines $lines, int $number): ?string
+    {
+        return null;
+    }
+
+    public function checkFileInfo(\SplFileInfo $fileInfo): ?string
+    {
+        $filename = u($fileInfo->getFilename());
+
+        if ($filename->containsAny('-')) {
+            return sprintf(
+                'Please don\'t use dashes (-) in filename: %s',
```

Comment:
The message is misleading. There should be hint what should be used instead

Reply:
Indeed, will fix that

Reply:
Done